File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string res) {
        int depth = 0;
        string s;
        for (int i = 0; i < res.length(); i++) {
            if (res[i] == '(') {
                depth++;
                if (depth > 1)
                    s += '(';
                
            } else {
                depth--;
                if (depth > 0)
                    s += ')';
            }
        }
        return s;
    
    }
};

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
       int b=0;
       String str="";
       for(int i=0;i<s.length();i++){
        if(s.charAt(i)=='('){
            b++;
            if(b>1){
                str+="(";
            }
        }
        else{
            b--;
            if(b>0){
                str+=")";
            }
        }
       
       } 
        return str;
    }
}